### PR TITLE
sprinkle some @suppress {missingProperties} for closure compilation

### PR DESCRIPTION
**Pull request by danbeam**
/cc @rictic

### CODE REVIEW

**Comment by rictic** [comment]
LGTM but deferring to Valdrin

File: iron-overlay-manager.html [hunk 1]
```diff
@@ -385,6 +386,7 @@
      * @param {!Element} overlay1
      * @param {!Element} overlay2
      * @return {boolean}
+     * @suppress {missingProperties}
```
Comment (valdrinkoshi) [inline iron-overlay-manager.html]: In the method we're already doing `/** @type {?} */ (overlay1)` for this reason, so we shouldn't need the `@suppress` here, right?

**Comment by valdrinkoshi** [comment]
@danbeam left a comment

File: iron-overlay-manager.html (hunk 1 above)
Comment (danbeam) [inline iron-overlay-manager.html]: if it was working, why would I need to send you a pull request? :wink: 

with the cast, chrome's code still fails to compile like this:
```
## /chromium/src/third_party/polymer/v1_0/components-chromium/iron-overlay-behavior/iron-overlay-manager-extracted.js:377: ERROR - Property alwaysOnTop never defined on o1
##       return !o1.alwaysOnTop && o2.alwaysOnTop;
##                  ^
```
with the suppression: compilation succeeds.

obviously somebody added this to fix their error, but Chrome likely has slightly different flag combo.

tl;dr - i can remove the cast if you'd like, but I don't understand who/how it's helping so didn't want to change it.

Comment (valdrinkoshi) [inline iron-overlay-manager.html]: Hehe yeah I guessed this is still creating problems, was interested to know more context, especially because I added that cast to specifically target that same error noticed in `source: hooli` 😁
May I ask you to remove the cast from this method please? 🙏

Comment (danbeam) [inline iron-overlay-manager.html]: Done. (removed cast)

**Comment by valdrinkoshi** [comment]
LGTM  :+1:
Thanks @danbeam!
